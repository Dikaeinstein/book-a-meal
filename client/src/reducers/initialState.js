const initialState = {
  urls: [
    {
      id: 1,
      name: 'Signin',
      link: 'signin',
    },
    {
      id: 2,
      name: 'Signout',
      link: 'signout',
    },
  ],
  menus: {
    isFetching: false,
    data: {
      menu: {
        name: 'Menu for today',
        meals: [],
      },
      menus: [],
    },
    isSet: false,
    isSaving: false,
    fetchError: null,
    saveError: null,
    updateError: null,
    isUpdating: false,
  },
  user: {
    isSubmitting: false,
    loggedIn: false,
    data: {
      id: null,
      name: '',
      email: '',
      role: '',
    },
    error: null,
  },
  orders: {
    isFetching: false,
    isSaving: false,
    isUpdating: false,
    isDeleting: false,
    data: {
      userOrders: [],
      checkedOutOrder: {},
    },
    fetchError: null,
    saveError: null,
    updateError: null,
    deleteError: null,
  },
  meals: {
    isFetching: false,
    isUpdating: false,
    isSaving: false,
    data: [],
    fetchError: null,
    saveError: null,
    updateError: null,
  },
  dashboard: {
    isFetchingTotalOrders: false,
    isFetchingTotalSales: false,
    isUpdating: false,
    isDeleting: false,
    fetchTotalOrdersError: null,
    fetchTotalSalesError: null,
    isFetchingAllOrders: false,
    fetchAllOrdersError: null,
    deleteOrderError: null,
    updateOrderError: null,
    totalAmount: 0,
    totalOrders: 0,
    allOrders: [],
  },
};

export default initialState;
