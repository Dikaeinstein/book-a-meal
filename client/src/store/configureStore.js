import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
