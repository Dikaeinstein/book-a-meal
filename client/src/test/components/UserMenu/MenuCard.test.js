import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedMenuCard, { MenuCard }
  from '../../../components/UserMenu/MenuCard';
import initialState from '../../../reducers/initialState';


describe('<MenuCard />', () => {
  let wrapper;
  beforeAll(() => {
    const mockStore = configureStore();
    const store = mockStore(initialState);
    wrapper = shallow(<ConnectedMenuCard
      link="/test"
      store={store}
      fetchMenu={jest.fn()}
      currentUrl="http://testurl"
    />);
  });
  it('should render with correct props', () => {
    expect(wrapper.props().link).toEqual('/test');
    expect(wrapper.props().isFetching).toBeFalsy();
    expect(wrapper.props().fetchMenu).toBeDefined();
  });
  it('should match snapshot', () => {
    let tree = shallow(<MenuCard
      link="/test"
      fetchMenu={jest.fn()}
      isFetching
      currentUrl="http://testurl"
    />);
    expect(tree).toMatchSnapshot();
    tree = shallow(<MenuCard
      link="/test"
      fetchMenu={jest.fn()}
      isFetching={false}
      currentUrl="http://testurl"
    />);
    expect(tree).toMatchSnapshot();
  });
});
