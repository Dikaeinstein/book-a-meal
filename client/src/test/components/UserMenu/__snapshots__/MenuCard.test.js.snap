// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuCard /> should match snapshot 1`] = `
<div
  className="loader-container"
>
  <Loading
    text="fetching menu..."
  >
    <Preloader
      active={true}
      color="blue"
      flashing={true}
      size="medium"
    />
  </Loading>
</div>
`;

exports[`<MenuCard /> should match snapshot 2`] = `
<section
  className="main"
  id="menu"
>
  <h2
    className="text-center"
  >
    Menu for Today 
    Mon Sep 03 2018
  </h2>
  <Connect(Component)
    link="/test"
  />
  <Paginate
    nextUrl=""
    onPageChange={[Function]}
    previousUrl=""
    style={
      Object {
        "marginTop": "0",
      }
    }
  />
</section>
`;
